### DISCUSSION TITLE
nnn preview-tui in wezterm

### DISCUSSION THREAD

**Discussion by kuntau** [wezterm/wezterm#3041] [Q&A] [answered]
I would like to add `wezterm` support in [preview-tui](https://github.com/jarun/nnn/blob/master/plugins/preview-tui) like kitty does.

Is it possible to do something like this? If it is possible, please point me to relevant docs. I tried searching but can't find relevant information.

**Comment by wez** [comment]
Probably a better question for the people that maintain preview-tui, as I don't use it and don't have time to investigate how it works.

wezterm supports all three image protocols: iTerm2, sixel and the kitty protocol.

The easiest way to preview an image in wezterm is to use `wezterm imgcat`. See https://wezfurlong.org/wezterm/imgcat.html

The kitty `icat` kitten should also work when run inside wezterm.

**Reply by kuntau** [reply]
Hi Wez, 

This is not so much about viewing images, it's general previewer. The problem to solve is creating a split and controlling that split with `nnn`. 

Kitty have `allow_remote_control` and `listen_on unix:$TMPDIR/kitty` settings, did wezterm have feature like this?

**Reply by wez** [reply]
https://wezfurlong.org/wezterm/cli/cli/index.html

**Reply by kuntau** [reply]
Thank you. Manage to get it working.

**Reply by kuntau** [reply]
> The kitty icat kitten should also work when run inside wezterm.

How does this work? I tried running `kitty +kitten icat IMG.jpg` but nothing happens.

**Reply by rafo** [reply]
> Thank you. Manage to get it working.

How? Could you share your solution?

**Comment by kuntau** [comment, marked as answer]
`nnn` `preview-tui` support in `wezterm` https://github.com/jarun/nnn/pull/1578
